src/main/java/org/variantsync/diffdetective/diff/difftree/DiffNode.java

```diff
@@ -1112,6 +764,204 @@ public String toTextDiff() {
+            // The method `DiffNode.addChild` can't be used here because `index` has a different
+            // meaning: For `DiffNode.addChild` it counts all children, before and after, but here
+            // it only counts children at `time`.
+
+            var iterator = getBackingNode().childOrder.listIterator();
+            for (int i = 0; i < index; ) {
+                if (!iterator.hasNext()) {
+                    throw new IllegalArgumentException();
+                }
+
+                if (iterator.next().getDiffType().existsAtTime(time)) {
+                    ++i;
+                }
+            }
+
+            iterator.add(child.getBackingNode());
+        }
+
+        @Override
+        public boolean removeChild(final VariationNode<DiffNode> child) {
+            return getBackingNode().removeChild(child.getBackingNode(), time);
+        }
+
+        @Override
+        public void removeAllChildren() {
+            getBackingNode().removeChildren(time);
+        }
+
+        @Override
+        public Node getDirectFeatureMapping() {
+            return getBackingNode().getDirectFeatureMapping();
+        }
+
+        @Override
+        public int getID() {
+            return getBackingNode().getID();
+        }
+
+        @Override
+        public boolean isSameAs(VariationNode<DiffNode> other) {
+            if (other != null && getClass() == other.getClass()) {
+                Projection otherProjection = (Projection) other;
+                return time.equals(otherProjection.time) && getBackingNode() == otherProjection.getBackingNode();
+            } else {
+                return false;
+            }
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Projection projection = (Projection) o;
+            return time.equals(projection.time) && getBackingNode().equals(projection.getBackingNode());
+        }
+    };
+
+    /**
+     * Returns a view of this {@code DiffNode} as a variation node at the time {@code time}.
+     *
+     * <p>See the {@code project} function in section 3.1 of
+     * <a href="https://doi.org/10.1145/3540250.3549108">our paper</a>.
+     */
+    public VariationNode<DiffNode> projection(Time time) {
+        Assert.assertTrue(getDiffType().existsAtTime(time));
+
+        if (projections[time.ordinal()] == null) {
+            projections[time.ordinal()] = new Projection(time);
+        }
+
+        return projections[time.ordinal()];
+    }
+
+    public static <T> DiffNode unchanged(VariationNode<T> variationNode) {
+        int from = variationNode.getFromLine();
+        int to = variationNode.getToLine();
+
+        var diffNode = new DiffNode(
+            DiffType.NON,
+            variationNode.getNodeType(),
+            new DiffLineNumber(from, from, from),
+            new DiffLineNumber(to, to, to),
```

Comment:
This method creates a `DiffNode` from a single `VariationNode` as if diffing the given variation node with itself. Hence, the name `unchanged`.
If the resulting `DiffNode` is printed as diff, there would be no added or removed lines. So the line numbers are all identical.
But I agree that this method needs documentation to clarify that.

Reply:
Ok I see. Then leave as is for now. One could insert the created DiffNode into a DiffTree though and thus, invalidate the line numbers.

Reply:
If you mean a different variation diff by *DiffTree*, then yes, but this would be reusing the nodes of one `DiffNode` tree in another (unconnected) `DiffNode` tree. This is always problematic for line numbers, regardless of how they are created.

Reply:
You are right, line numbers get always messed up by tree manipulations. Then keep as is. But I was referring to a DiffTree by DiffTree because a DiffTree is made of DiffNodes, so a DiffNode could be inserted into a DiffTree.